src/s1reader/s1_burst_slc.py

```diff
@@ -781,7 +787,93 @@ def az_fm_rate_mismatch_mitigation(self,
+        proj = isce3.core.make_projection(epsg)
+        ellipsoid = proj.ellipsoid
+        grid_doppler = isce3.core.LUT2d()
+
+        rdr2geo_obj = isce3.geometry.Rdr2Geo(
+                correction_radargrid,
+                self.orbit,
+                ellipsoid,
+                grid_doppler,
+                threshold=threshold_rdr2geo,
+                numiter=numiter_rdr2geo)
+
+        str_datetime = datetime.datetime.now().strftime('%Y%m%d_%H%M%S%f')
+        list_filename_llh = [f'{path_scratch}/{llh}_{str_datetime}.rdr'
+                             for llh
+                             in ['lat', 'lon', 'hgt']]
+        lat_raster, lon_raster, hgt_raster = [isce3.io.Raster(
+                                                        filename_llh,
+                                                        correction_radargrid.width,
+                                                        correction_radargrid.length,
+                                                        1,
+                                                        gdal.GDT_Float64,
+                                                        'ENVI')
+                                              for filename_llh in list_filename_llh]
+
+        rdr2geo_obj.topo(dem_raster, lon_raster, lat_raster, hgt_raster)
+
+        # make sure that the ISCE3 rasters are written out to file system
+        lat_raster.close_dataset()
+        lon_raster.close_dataset()
+        hgt_raster.close_dataset()
+
+        # Load the lon / lat / hgt value from the raster
+        arr_lat = gdal.Open(list_filename_llh[0], gdal.GA_ReadOnly).ReadAsArray()
+        arr_lon = gdal.Open(list_filename_llh[1], gdal.GA_ReadOnly).ReadAsArray()
+        arr_hgt = gdal.Open(list_filename_llh[2], gdal.GA_ReadOnly).ReadAsArray()
+
+        lut_az_fm_mismatch = self.az_fm_rate_mismatch_from_llh(arr_lat,
+                                                               arr_lon,
+                                                               arr_hgt,
+                                                               ellipsoid,
+                                                               correction_radargrid)
+
+        # Clean up the temporary directory in case it exists.
+        if temp_dir_obj is not None:
+            temp_dir_obj.cleanup()
+
+        return lut_az_fm_mismatch
+    
+
+    def az_fm_rate_mismatch_from_llh(self,
+                                     lat_map: np.ndarray,
+                                     lon_map: np.ndarray,
+                                     hgt_map: np.ndarray,
+                                     ellipsoid: isce3.core.Ellipsoid,
+                                     correction_radargrid: isce3.product.RadarGridParameters):
+        '''
+        Take in lat / lon / hgt in radar grid, along with ellipsoid and the radar grid.
+        Calculate azimuth FM rate mismatch mitigation
+        Based on ETAD-DLR-DD-0008, Algorithm Technical Baseline Document.
```

Comment:
I'd like to keep citation to references consistent with the numpy docstring. See Sec. 14 here https://numpydoc.readthedocs.io/en/latest/format.html